### ISSUE TITLE
[BUG] Val'anyr, Hammer of Ancient Kings Producing Infinitely Large Absorption Shields

### ISSUE THREAD

**Issue by CharWS** [JamminL/wotlk-classic-bugs#634]
#### 1.2. How to Reproduce
           1. Proc Blessing of Ancient Kings
           2. Heal a target such that they have an absorb shield in excess of 20,000
           3. Check how much damage is absorbed by the shield before it expires (noting that it can far exceed 20,000)
  #### 1.3  Source Material
https://streamable.com/hnmvau in this video can see that 158,370 healing is done for a total shield quantity of 23,755.5 and that the damage absorbed is far in excess of that amount
https://streamable.com/5fzqu0 in this video can see that 121,257 healing is done for a total shield quantity of 18,188.55 and that the shield breaks a few seconds prior to its expiration timer due to that absorb amount being used up
#### 2.2 Source Material
In this bluepost it's noted that the maximum absorb shield should cap out at 20,000 damage (before some is removed and additional heals are applied) https://www.bluetracker.gg/wow/topic/us-en/16903512910-valrsquoanyr-hammer-of-the-ancient-kings/
While it's possible that a later undocumented change could have raised the cap beyond 20,000 it still should be limited to only absorbing 15% of the healing done while under Blessing of Ancient Kings.

**Comment by Aggrend** [comment]
So this is a pretty hilarious, yet completely reference accurate bug.

TLDR, there's a very simple statement in the spell script that is intended to cap the shield at 20k, but someone made a typo and it caps at 200k.

Something like:
```
if (amount>20000) then
        amount=200000
    end
```

This was fixed in Mists of Pandaria, and was simply never noticed until then. We are fixing it to correctly cap at 20k in Wrath Classic. This has been hotfixed as of about 5 minutes ago. Within a few hours this should be totally resolved on live.

**Comment by MghTG** [comment]
Lol
That explains a video I found some time ago of a tanker absorving two entire hits of 40k+ dmg on Lich King 25H.
